### DISCUSSION TITLE
Index of all Challenges

### DISCUSSION THREAD

**Discussion by npa528** [Cyfrin/foundry-full-course-cu#135] [General]
I thought it would be nice to have all Challenges in a one place. I don't find QR codes very convenient so I decided to put them all together here.

Note: I opened an issue that Challenge 12 was missing from the videos (QR) but I found that there is actually a Challenge 12 so I added this in this list. 

Challenge / Lesson | Blockchain | URL
-- | -- | --
1 | Arbitrum           Sepolia | https://arbiscan.io/address/0x2e99A4CA9c85383ccB54769adB3837437e296479#code       https://sepolia.etherscan.io/address/0x25056312685339b49e1d1C5a0b72Ff9eff13AF77#code
2 | Arbitrum           Sepolia | https://arbiscan.io/address/0x76D2403b80591d5F6AF2b468BC14205fa5452AC0#code    https://sepolia.etherscan.io/address/0x5c1ddb86F11BB46D3067C702AC554aEaED9ff8f0#code
3 | Arbitrum           Sepolia | https://arbiscan.io/address/0x34d130B174F4a30A846FED7C02FCF53A19a4c2B6#code    https://sepolia.etherscan.io/address/0x4A62A8901e6113dEF0AEeEC77E531779cd40da73#code
4 | Arbitrum           Sepolia | https://arbiscan.io/address/0xA2626bE06C11211A44fb6cA324A67EBDBCd30B70#code    https://sepolia.etherscan.io/address/0x1b30DA2a868704483143A4D46865Ac9585629fd0#code
5 | Arbitrum           Sepolia | https://arbiscan.io/address/0xf988Ebf9D801F4D3595592490D7fF029E438deCa#code    https://sepolia.etherscan.io/address/0x4b3a7F293091708dDD6B8748179aeAF80E9c1bA2#code
6 | Arbitrum           Sepolia | https://arbiscan.io/address/0xdeB8d8eFeF7049E280Af1d5FE3a380F3BE93B648#code    https://sepolia.etherscan.io/address/0x6c4791c3a9E9Bc5449045872Bd1b602d6385E3E1#code
7 | Arbitrum           Sepolia | https://arbiscan.io/address/0xcf4fbA490197452Bd414E16D563623253eFb57D3#code    https://sepolia.etherscan.io/address/0xD7D127991c6A89Df752FC3daeC17540aE8B86101#code
8 |   | ipfs://QmUgc1xpNgTMwSF2GBS5FTFKmSdrmSnLJD5DoYZtcQ2n5X
9 | Arbitrum               Sepolia | https://arbiscan.io/address/0xdF7cdFF0c5e85c974D6377244D9A0CEffA2b7A86#code    https://sepolia.etherscan.io/address/0x33e1fD270599188BB1489a169dF1f0be08b83509#code
10 | Arbitrum           Sepolia | https://arbiscan.io/address/0xaB67557218F60C06acA750B9F8A20018e5604ebf#code    https://sepolia.etherscan.io/address/0xE0aE410a16776BCcb04A8d4B0151Bb3F25035994#code
11 | Arbitrum           Sepolia | https://arbiscan.io/address/0x444aE92325dCE5D14d40c30d2657547513674dD6#code    https://sepolia.etherscan.io/address/0x93c7A945af9c453a8c932bf47683B5eB8C2F8792#code
12 | Arbitrum           Sepolia | https://arbiscan.io/address/0x3dbbf2f9acfb9aac8e0b31563dd75a2d69148d64#code       https://sepolia.etherscan.io/address/0xe5760847db2f10a74fc575b4803df5fe129811c1#code
13 | Arbitrum           Sepolia | https://arbiscan.io/address/0x3aD9F29AB266E4828450B33df7a9B9D7355Cd821#code    https://sepolia.etherscan.io/address/0xaFa4150818b7843345A5E54E430Bd0cAE31B5c0C#code
14 | Arbitrum           Sepolia | https://arbiscan.io/address/0xc584bD01fD60F671409661a6802170BbEFba5c47#code    https://sepolia.etherscan.io/address/0x46F3fE2C8aC9e9AE4DEDE1a7a29Ab3BdcFa7eaFc#code
15 | Arbitrum           Sepolia | https://arbiscan.io/address/0xa0c7ADA2c7c29729d12e2649BC6a0a293Ac46725#code    https://sepolia.etherscan.io/address/0x766a74f8924C7B07df088fDB0F7D7DbaDd330Fb3#code

**Comment by devblixt** [comment]
Great work!

**Reply by RichardPalladino** [reply]
You can also go to the NFT contracts themselves and lookup each challenge by index (lesson 1 is index 0, lesson 2 challenge has index 1, etc.).

[https://sepolia.etherscan.io/token/0x76b50696b8effca6ee6da7f6471110f334536321#readContract](https://sepolia.etherscan.io/token/0x76b50696b8effca6ee6da7f6471110f334536321) or  [https://arbiscan.io/token/0x39338138414df90ec67dc2ee046ab78bcd4f56d9#readContract](https://arbiscan.io/token/0x39338138414df90ec67dc2ee046ab78bcd4f56d9) and call the `getChallengeContract()` function.

**Reply by PatrickAlphaC** [reply]
niceeeeeee

**Comment by MotiMorosi** [comment]
So nice!

**Comment by Nisogok** [comment]
Thank you very much, that's really helpful.

**Comment by abdoMous** [comment]
I've noticed that the course hasn't specified the challenge for Lesson 12. Is that correct?

**Reply by Niferu** [reply]
Seems like QR code is missing for it indeed, but if you were doing all previous challenges you can go into next by index as [RichardPalladino](https://github.com/RichardPalladino) mentioned.

**Reply by abdoMous** [reply]
Thanks, @Neftyr! I noticed that the QR code for Lesson 12 has been added to the repository's [README](https://github.com/Cyfrin/foundry-full-course-f23#more-defi-learnings).
